src/js/state/migrations/202009121941_refactorInvestigations.ts

```diff
@@ -0,0 +1,19 @@
+import {getAllStates} from "../../test/helpers/getTestState"
+
+export default function refactorInvestigations(state: any) {
+  for (const s of getAllStates(state)) {
+    const newState = {}
+    s.investigation.forEach((f) => {
+      if (!f) return
+      if (!newState[f.search.spaceId]) newState[f.search.spaceId] = []
+
+      newState[f.search.spaceId].push(f)
+    })
+
+    s.investigation = {
```

Comment:
Since we don't know the connId associated with a given finding, the *best we can do is just put them all on `localhost:9867` and hope that since the new connection feature is so new that folks haven't yet leveraged it :( 

*this got me thinking, @jameskerr we had discussed organizing the state as I've done in these commits 
```
"investigations": {
    "connId": {
      "spaceId": [{finding}]
    }
}
```
...which allows us to remove all of a connection's spaces' investigations more simply/quickly, BUT it creates this migration problem AND given the way we are generating spaceIds we should actually be able to rely on them all being unique across connections anyway.

So now I'm thinking of refactoring the state to the following, and on removing a connection we instead pay the cost of removing each of the connection's spaces/investigations sequentially

```
"investigations": {
  "spaceId": [{finding}]
}
```

wdyt?

Reply:
We decided offline to keep as is for the benefits mentioned above, and assume that folks have not created new connections